### PR TITLE
PWGHF: Shift decay type values to allow usage in arrays and bitmaps.

### PR DESCRIPTION
**Pull request by vkucera**
This allows to use the `enum` values of decay types as indices to access array elements and as bit positions in candidate selection bitwise operations.
Related PRs: https://github.com/AliceO2Group/AliceO2/pull/5218 https://github.com/AliceO2Group/AliceO2/pull/5223

### CODE REVIEW

**Review by jgrosseo — APPROVED** [review]

**Comment by jgrosseo** [comment]
Please wait for the approval of @ginnocen

**Review by ginnocen — APPROVED** [review]

**Pull request merged**
